src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs

```diff
@@ -1231,6 +1231,9 @@ void M4(int* p)
                 // (6,29): error CS8521: Pattern-matching is not permitted for pointer types.
                 //     bool M1(int* p) => p is null; // 1
                 Diagnostic(ErrorCode.ERR_PointerTypeInPatternMatching, "null").WithLocation(6, 29),
+                // (7,24): warning CS8794: An expression of type 'int*' always matches the provided pattern.
```

Comment:
nit: should/could we try and avoid the cascade here? The error (not pattern-matching on pointer type) seems more fundamental